### ISSUE TITLE
[BUG] mini12864 with Fysetc S6 shifts display contents up after about 30s

### ISSUE THREAD

**Issue by Drachenkaetzchen** [MarlinFirmware/Marlin#16059]
### Bug Description

After about 30 seconds, the fysetc mini12864 display contents gets shifted up by about half a screen, but continues to function. When returning to the info screen and entering the menu again, the contents are displayed as expected again until 30 seconds later.

The same display works fine on Rumba and SKR 1.3 boards, so I assume it's an issue specific to the Fysetc S6.

### My Configurations

[Configuration.h](https://github.com/Drachenkaetzchen/Marlin/blob/PurpleCube-S6/Marlin/Configuration.h)
[Configuration_adv.h](https://github.com/Drachenkaetzchen/Marlin/blob/PurpleCube-S6/Marlin/Configuration_adv.h)

### Steps to Reproduce

Use a fysetc S6 connected to a fysetc mini12864 display. I used about 30cm long cables. Insert at least 3 TMC2130 drivers in SPI mode. It doesn't matter if the TMC drivers are enabled in the config.

1. Wait about 30 seconds
2. Observe that the display contents have shifted

I verified that the SW does not send the start line command, so it's possibly due to noise on the data line. `FORCE_SOFT_SPI` results in a blank display. The SPI speed is about 3.2MHz, and the line shifting did occur even with `SPI_SPEED_6`.

#### Additional Information
![IMG_20191129_195435](https://user-images.githubusercontent.com/162974/69906507-dcf84f80-13c4-11ea-8ab1-01145ead86bf.jpg)

**Comment by Drachenkaetzchen** [comment]
I tested with a Reprapdiscount Smart Controller, it does not show that behavior.

**Comment by Drachenkaetzchen** [comment]
I also increased the display timings, no effect:

```
#define BOARD_ST7920_DELAY_1 DELAY_NS(206)
#define BOARD_ST7920_DELAY_2 DELAY_NS(102)
#define BOARD_ST7920_DELAY_3 DELAY_NS(1284)
```

**Comment by Drachenkaetzchen** [comment]
I have had a further look at this and the Fysetc mini12864 uses SPI. The datasheet for the ST7567 
 controller specifies a maximum of 20MHz for the clock. Gotta get my scope out and see how fast it actually is.

**Comment by Drachenkaetzchen** [comment]
Also would be interesting to try `FORCE_SOFT_SPI` to see if it makes any difference. Not sure if it's a timing issue though. The shifting seems too consistent.

**Comment by Drachenkaetzchen** [comment]
More test results:

- Issue occurs with 3 or more TMC2130 drivers plugged in. It doesn't seem to matter if the stepstick position is actually enabled. For example, I have X,Y,Z,E0 enabled as TMC2130 and put TMC2130 drivers in X,Y and E1. Issue occurs.
- Issue doesn't seem to occur with only 2 TMC2130 drivers plugged in

**Comment by Drachenkaetzchen** [comment]
- `FORCE_SOFT_SPI` results in a blank display, didn't mess around with that any further
- Disabling the driver types or reverting to `A4988` doesn't do anything
- The Fysetc fork of Marlin shows similar behavior

Right now I can't explain the issue differently than some weird signal reflections.

**Comment by Drachenkaetzchen** [comment]
The measured SPI Clock is about 1.4MHz.

**Comment by Drachenkaetzchen** [comment]
Looking through the `u8g2` code it seems that the `ST7565_START_LINE` is only used during initialization. So if there's noise or reflections on the line and the display interprets the `ST7565_START_LINE` by that, the display gets shifted but never reverts back to start line 0, unless the user switches from the status screen to the main menu, where the display is actually reset every time, which in turn reverts the shifted display. To me that's the most plausible explanation right now.

**Comment by Drachenkaetzchen** [comment]
Potentially related fixes:

https://github.com/MarlinFirmware/Marlin/commit/23bb40d0ea398cee3f68bd9a4a716a3b7dda8db9
https://github.com/MarlinFirmware/Marlin/issues/13550

**Comment by Drachenkaetzchen** [comment]
I have to stop debugging this as the TMC drivers are needed in my other machine. I updated the initial report with my findings

**Comment by iankds05** [comment]
For what it is worth one of the issues I have found with RepRap type displays is that they are much more susceptible to jitter and noise in the ribbon cables connected to 32bit MPU boards than they ever were on 8bit ones, especially if run in proximity to any other cables, either in or outside of a harness. On 8bit Atmel 2560 boards I have run the displays over a meter away from the main control board with no issues but with 32bit MPU boards anything over 150mm on the ribbon cables causes display disfunction either in the form of scrambled pixels or jumpy response to menu selections. One successful solution I have found is to abandon the ribbon cables, and customise new cables using 28AWG single cores which still fit into the IDC 10 way connectors and by twisting the 10 cores together and placing a ferrite core on one end nearest the display all issues seem to be eliminated. Weather this would be a possible solution to this particular issue I cannot be sure as I don't posses the afore mentioned display to test it thoroughly

**Comment by AnHardt** [comment]
It's not the difference 8 to 32 bit. It's 5 vs. 3.3V, lowering the noise distance and making the ST7920 even slower (see [datasheet](http://www.hpinfotech.ro/ST7920.pdf) Page 43-44).

**Comment by jonathanlundstrom** [comment]
Has this problem been fixed in Marlin 2.0.5.2?

**Comment by jonathanlundstrom** [comment]
This problem is still highly prominent. I also noticed that if you've got the speaker/buzzer enabled in `Configuration.h` then motor movements will not work properly. This will also cause the jitter to increase severely.

While debugging yesterday I also ran into another problem. When printing over SD the printer just decides to skip chunks of code which results in some very strange results. I've tried several models through several SD-cards but the problem only occurs though SD. It works fine over USB.
